lib/simplekiq/orchestration_executor.rb

```diff
@@ -12,35 +12,40 @@ def self.execute(args:, job:, workflow:)
       orchestration_batch.description = "#{job.class.name} Simplekiq orchestration"
       Simplekiq.auto_define_callbacks(orchestration_batch, args: args, job: job)
 
-      new.run_step(orchestration_batch, workflow, 0)
+      orchestration_batch.jobs do
+        Simplekiq::BatchTrackerJob.perform_async(job.class.name, orchestration_batch.bid, args)
+
+        new.run_step(workflow, 0)
+      end
     end
 
-    def run_step(orchestration_batch, workflow, step)
+    def run_step(workflow, step)
       *jobs = workflow.at(step)
       # This will never be empty because Orchestration#serialized_workflow skips inserting
       # a new step for in_parallel if there were no inner jobs specified.
 
-      orchestration_batch.jobs do
-        step_batch = Sidekiq::Batch.new
-        step_batch.on(
-          "success",
-          self.class,
-          {"orchestration_workflow" => workflow, "step" => step + 1}
-        )
-
-        step_batch.jobs do
-          jobs.each do |job|
-            Object.const_get(job["klass"]).perform_async(*job["args"])
-          end
+      next_step = step + 1
```

Comment:
longer term consider abstracting this into a counter object, it can have validation logic (since it's coming from an outside source) and implement logic to limit the number of steps down if performance becomes a concern.